### PR TITLE
[mesheryctl] Add support for EKS config

### PR DESCRIPTION
**Pull request by hexxdump**
[mesheryctl] Add support for EKS config

Signed-off-by: Hussaina Begum Nandyala <hexxdump@gmail.com>

**Description**
Add support for "mesheryctl system config eks"

This PR fixes #

**Notes for Reviewers**
Could not test the changes as I do not have EKS cluster.

**[Signed commits](https://github.com/layer5io/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Yay!

**Comment by leecalcote** [comment]
Yay!

File: mesheryctl/pkg/utils/scripts.go [hunk 1]
```diff
@@ -162,3 +162,44 @@ func GenerateConfigAKS(resourceGroup, clusterName string) error {
...
 }
+
+
+// GenerateConfigEKS generates kube config file in /tmp/meshery/kubeconfig.yaml for a GKE cluster
+func GenerateConfigEKS(region, cluster string) error {
+	script := fmt.Sprintf(`
+	set -e
+	set -o pipefail
+
+	TARGET_FOLDER="/tmp/meshery/"
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:173]: @hexxdump can we avoid use of a temp location and use Meshery's application folder instead? `~/.meshery`

Comment (hexxdump) [reply mesheryctl/pkg/utils/scripts.go]: Done @leecalcote

Comment (ameyaagashe) [reply mesheryctl/pkg/utils/scripts.go]: Looking fine to me after the directory update. Need to test it out!

File: mesheryctl/pkg/utils/scripts.go [hunk 2]
```diff
@@ -162,3 +162,44 @@ func GenerateConfigAKS(resourceGroup, clusterName string) error {
...
 }
+
+
+// GenerateConfigEKS generates kube config file in /tmp/meshery/kubeconfig.yaml for a GKE cluster
+func GenerateConfigEKS(region, cluster string) error {
+	script := fmt.Sprintf(`
+	set -e
+	set -o pipefail
+
+	TARGET_FOLDER="~/.meshery/"
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:173]: @hexxdump can we avoid creating a new directory? We can reuse the user's existing `MesheryFolder` (https://github.com/layer5io/meshery/blob/master/mesheryctl/pkg/utils/helpers.go#L63).

File: mesheryctl/pkg/utils/scripts.go [hunk 3]
```diff
@@ -162,3 +162,44 @@ func GenerateConfigAKS(resourceGroup, clusterName string) error {
...
+	KUBECFG_FILE_NAME="$TARGET_FOLDER/kubeconfig.yaml"
+	KUBECONFIG=${KUBECFG_FILE_NAME}
+
+	REGION_NAME=%s
+	CLUSTER_NAME="%s"
+
+	create_target_folder() {
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:180]: Remove this function?

File: mesheryctl/pkg/utils/scripts.go [hunk 4]
```diff
@@ -162,3 +162,44 @@ func GenerateConfigAKS(resourceGroup, clusterName string) error {
...
+		echo -n "Creating target directory to hold files in ${TARGET_FOLDER}..."
+		mkdir -p "${TARGET_FOLDER}"
+		printf "done"
+	}
+
+	create_update_kubeconfig() {
+		echo -e "\\nQuerying the kubernetes configuration for cluster ${CLUSTER_NAME} from EKS..."
+		aws eks --region ${REGION_NAME} update-kubeconfig --name ${CLUSTER_NAME} --kubeconfig ${KUBECONFIG}
```
Comment (leecalcote) [inline mesheryctl/pkg/utils/scripts.go:188]: We need to try / catch when using `aws`. Let's first check for the availability of this requisite dependency.

File: mesheryctl/pkg/utils/scripts.go (hunk 2 above)
Comment (leecalcote) [reply mesheryctl/pkg/utils/scripts.go:173]: This portion of the command can be done in golang. Let's put as little as possible into shell script.

**Comment by leecalcote** [comment]
Thank you, @hexxdump

**Review by leecalcote — APPROVED** [review]

**Pull request merged**